### DISCUSSION TITLE
Doubts about using sidekiq-ent / sidekiq-pro

### DISCUSSION THREAD

**Discussion by felixcuello** [sidekiq/sidekiq#5088] [Q&A] [answered]
Hello 👋 

We have recently purchased a `sidekiq` enterprise license in our company and my job is to put that to work but I encountered a few doubts that were partially answered by googling.

**CONTEXT**
In our organization we use `artifactory` as an intermediate cache for Ruby Gems and other kind of images that are external to our company. This is mostly to detach our builds from external dependencies, so if the external dependency is not working, we still can build our images.

This cache is used by all our environments _development_ , _QA_, _staging_ and _production_

**PROBLEM**
Due to this I have a few questions / comments that are bothering me to complete this work correctly.

1. **CACHE**: Can this cache interfere with the license?, because `artifactory` will download the gem and it will serve to all our environments.  **How the licensing system determines that something is running in DEV, QA, PROD**?, because we don't want to consume these licenses with the developer's computers (i.e. I think it count threads)
2. **LICENSING**: If the license can't be used in all environments. Is there any way to disable it in lower environments?   I was thinking to put a _feature switch_ , so it loads the `sidekiq` or `sidekiq-ent` gem, but there are some methods not supported by the regular gem, that are available in the pro versions, also I saw there's a `.pro?` method or also I saw that the recommendation 4 years ago was to ise `defined?(Sidekiq::Pro)` **Are these the only ways to do it?**

Thank you very much indeed for all the help.

Cheers

**Comment by mperham** [comment, marked as answer]
You are welcome to use a gem cache like Artifactory as long as it is truly a cache with a TTL of no more than a few days.

The environment is determined with `-e production` or `RAILS_ENV=production`.
